universal-profiling-integration/src/main/java/co/elastic/otel/UniversalProfilingProcessor.java

```diff
@@ -249,6 +268,20 @@ private void consumeProfilerMessages() {
     }
   }
 
+  private void handleMessage(ProfilerRegistrationMessage message) {
+    log.log(
+        Level.FINE,
+        "Received profiler registration message! host.id is {0} and the span delay is {1} ms",
+        new Object[] {message.getHostId(), message.getSamplesDelayMillis()});
+
+    tlsPropagationActive = true;
+    long spanDelayNanos =
+        Duration.ofMillis(message.getSamplesDelayMillis() + POLL_FREQUENCY_MS).toNanos();
+    correlator.setSpanDelayNanos(spanDelayNanos);
```

Comment:
Here it's a good place where the "delay" provided by the profiler get mixed with the known polling frequency to produce a "time buffer" for which we need to buffer spans.

Also, would it be relevant to have the profiler also provide the sampling frequency rather than hard-coding it in the agent or is it something very unlikely to change on the profiler side ?

Reply:
The `POLL_FREQUENCY_MS` is actually unrelated to the sampling frequency of the profiler.
This duration just defined how frequently we look into the profiling socket to check for messages.

Therefore I don't see any need to make this configurable right now. If the need arises, we can add it later.